frame/tips/src/lib.rs

```diff
@@ -599,4 +607,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Tips::<T, I>::insert(hash, new_tip)
 		}
 	}
+
+	#[cfg(any(feature = "try-runtime", feature = "fuzzing", test, debug_assertions))]
+	pub fn do_try_state() -> Result<(), &'static str> {
+		let reasons = Reasons::<T, I>::iter_keys().collect::<Vec<_>>();
+		let tips = Tips::<T, I>::iter_keys().collect::<Vec<_>>();
+		assert_eq!(reasons.len(), tips.len());
```

Comment:
Please document what is the invariant being held in the docs briefly? would help. 

Also, if the keys are the same, you can actually assert that they are actually the same, not just length.

Reply:
Will document the invariant for sure.

 > Also, if the keys are the same, you can actually assert that they are actually the same, not just length.
 
 noted

Reply:
you can still iterate all the reasons and ensure there is a matching tip